test/elixir_auth_google_test.exs

```diff
@@ -65,10 +74,35 @@ defmodule ElixirAuthGoogleTest do
       picture: "https://lh3.googleusercontent.com/a-/AAuE7mApnYb260YC1JY7a",
       sub: "940732358705212133793"
     }
+
     assert ElixirAuthGoogle.get_user_profile("123") == {:ok, res}
   end
 
   test "return error with incorrect token" do
-    assert ElixirAuthGoogle.get_user_profile("wrong_token") == {:error, :bad_request }
+    assert ElixirAuthGoogle.get_user_profile("wrong_token") == {:error, :bad_request}
+  end
+
+  test "generate_redirect_uri(conn) generate correct callback url" do
+    conn = %{
+      host: "foobar.com",
+      port: 80
+    }
+
+    assert ElixirAuthGoogle.generate_redirect_uri(conn) ==
+             "https://foobar.com/auth/google/callback"
+  end
+
+  test "generate_redirect_uri(conn) generate correct callback url with custom url path" do
+    conn = %{
+      host: "foobar.com",
+      port: 80
+    }
+
+    mock_app = [get_env: fn :elixir_auth_google, :callback_path -> "/special/callback" end]
+
+    with_mock Application, mock_app do
```

Comment:
do we _need_ to import a mocking library to do this? 💭

Reply:
Mmm I don't know another way to do it without mock... Maybe by setting this config in conf/test.exs?

On the other hand, the library is only for tests `{:mock, "~> 0.3.0", only: :test},`

Reply:
We could add the config to avoid needing the `mock` lib. 
But in fairness, the library looks well-maintained and documented https://github.com/jjh42/mock 
so let's just merge and if we decide to refactor it out later we can. 💭